Add .devcontainer Configuration

BrunoZell opened the pull request:
**Description**

This pull request introduces a `.devcontainer` configuration to the default Euclid repository:

- **Quick Setup**: New contributors can get started with a fully configured development environment in minutes, directly within VS Code.
- **Enhanced Consistency**: Ensures that all developers work in an identical development environment, reducing "works on my machine" issues.
- **ZSH as Default Shell**: Currently includes ZSH. Based on team feedback, we might consider switching to the less opinionated default shell.

Please review the proposed setup and share feedback regarding the choice of the shell and any other configurations.

Greetings from ZuBerlin 🌻

scasplte2 commented:
Following the instructions in the readme, I am able to open the devcontainer via VSCode as expected. Nicely done :fire: 

Probably worth getting input from @AlexBrandes & @IPadawans as well.

IPadawans commented:
Hey @BrunoZell I've tried to run according to README on my Mac and I'm facing some issues:
```
ERROR: failed to solve: process "/bin/sh -c apt-get update &&     apt-get install -y apt-transport-https ca-certificates curl software-properties-common lsb-release &&     curl -fsSL https://download.docker.com/linux/ubuntu/gpg | apt-key add - &&     add-apt-repository \"deb [arch=amd64] https://download.docker.com/linux/ubuntu $(lsb_release -cs) stable\" &&     apt-get update &&     apt-get install -y docker-ce-cli" did not complete successfully: exit code: 100

View build details: docker-desktop://dashboard/build/desktop-linux/desktop-linux/asq7goy9ehls0snxwilzriaqd
[55051 ms] Error: Command failed: docker buildx build --load --build-arg BUILDKIT_INLINE_CACHE=1 -f /var/folders/73/fl2vql2x2d3940p82pq140y80000gn/T/devcontainercli/container-features/0.62.0-1718216344021/Dockerfile-with-features -t vsc-euclid-development-environment-9b9d0d33c7dae62c8e210b384588cab7ddfed4080bd639c7be161595e7504fc7 --target dev_containers_target_stage --build-arg _DEV_CONTAINERS_BASE_IMAGE=dev_container_auto_added_stage_label /Users/marcusviniciusgirolnetosousa/Documents/repositories/engineer/euclid-development-environment/.devcontainer
[55051 ms]     at mtA (/Users/marcusviniciusgirolnetosousa/.vscode/extensions/ms-vscode-remote.remote-containers-0.369.0/dist/spec-node/devContainersSpecCLI.js:465:1933)
[55051 ms]     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
[55051 ms]     at async Pm (/Users/marcusviniciusgirolnetosousa/.vscode/extensions/ms-vscode-remote.remote-containers-0.369.0/dist/spec-node/devContainersSpecCLI.js:464:1841)
[55051 ms]     at async NH (/Users/marcusviniciusgirolnetosousa/.vscode/extensions/ms-vscode-remote.remote-containers-0.369.0/dist/spec-node/devContainersSpecCLI.js:464:610)
[55051 ms]     at async KtA (/Users/marcusviniciusgirolnetosousa/.vscode/extensions/ms-vscode-remote.remote-containers-0.369.0/dist/spec-node/devContainersSpecCLI.js:481:3692)
[55051 ms]     at async eB (/Users/marcusviniciusgirolnetosousa/.vscode/extensions/ms-vscode-remote.remote-containers-0.369.0/dist/spec-node/devContainersSpecCLI.js:481:4807)
[55052 ms]     at async hrA (/Users/marcusviniciusgirolnetosousa/.vscode/extensions/ms-vscode-remote.remote-containers-0.369.0/dist/spec-node/devContainersSpecCLI.js:661:13373)
[55052 ms]     at async lrA (/Users/marcusviniciusgirolnetosousa/.vscode/extensions/ms-vscode-remote.remote-containers-0.369.0/dist/spec-node/devContainersSpecCLI.js:661:13114)
[55061 ms] Exit code 1
```

Have you tested with Mac's ARM64 and AMD64?

BrunoZell commented:
> Have you tested with Mac's ARM64 and AMD64?

Good point - no I haven't! Please check the latest version. I added an argument `ARCH` to the Dockerfile which is automatically set by VS Code when building the Dev Container. I changed the installation of Docker Compose, Cursier, and YQ to install the correct binaries for the host architecture.

I don't have access to a Mac though so I can't test it. Might you want to run it on your Mac @IPadawans?
